content/blog/exposing-an-application-using-ingress-and-tls-termination-on-kubernetes-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,528 @@
+spec:
+  ingressClassName: nginx
+  tls:
+  - hosts:
+    - nginx.example.com
+    secretName: cfe-tls-key-pair
+  rules:
+  - host: nginx.example.com
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: nginx-main
+            port:
+              number: 80
+  - host: blue.nginx.example.com
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: nginx-blue
+            port:
+              number: 80
+  - host: green.nginx.example.com
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: nginx-green
+            port:
+              number: 80
+```
+
+In the above sample YAML manifest file, there is the *'tls'* block that contains the hostname *'nginx.example.com'* and the secret *cfe-tls-key-pair* created in the certification step. There is also the *'rules'* block in which a list of routing rules is defined per host, e.g., host *nginx.example.com* will be routed to the application service *nginx-main* in the backend.  
+
+T﻿ype the following command to deploy the Ingress resource to the namespace *nginx-apps*:
+
+```shell
+$ kubectl apply -f ingress-host-based-selfsigned.yaml -n nginx-apps
+ingress.networking.k8s.io/ingress-host-based-selfsigned created
+```
+
+Check the details of the *TLS* and *Rules* settings by t﻿yping below command:
+
+```shell
+$ kubectl describe ingress ingress-host-based-selfsigned -n nginx-apps
+Name:             ingress-host-based-selfsigned
+Labels:           <none>
+Namespace:        nginx-apps
+Address:
+Ingress Class:    nginx
+Default backend:  <default>
+TLS:
+  cfe-tls-key-pair terminates nginx.example.com
+Rules:
+  Host                     Path  Backends
+  ----                     ----  --------
+  nginx.example.com
+                           /   nginx-main:80 (10.192.4.44:80)
+  blue.nginx.example.com
+                           /   nginx-blue:80 (10.192.3.78:80)
+  green.nginx.example.com
+                           /   nginx-green:80 (10.192.4.45:80)
+Annotations:               cert-manager.io/issuer: nginx-selfsinged-issuer
+                           ingress.kubernetes.io/ssl-redirect: true
+Events:
+  Type    Reason             Age   From                       Message
+  ----    ------             ----  ----                       -------
+  Normal  Sync               20s   nginx-ingress-controller   Scheduled for sync
+  Normal  CreateCertificate  20s   cert-manager-ingress-shim  Successfully created Certificate "cfe-tls-key-pair"
+```
+
+### Access deployed Nginx applications
+
+B﻿efore start accessing the deployed Nginx applications, you need set up the domain and the subdomain name resolution. For the sample domain name *nginx.example.com*, and its subdomains, *blue.nginx.example.com* and *green.nginx.example.com*, the workstation host file has been used for DNS resolution.  
```

Comment:
Typo in the first sentence:
Before accessing the deployed Nginx applications...